fleetspeak/src/windows/regutil/regutil_windows.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package regutil
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sys/windows/registry"
+)
+
+const hkeyPrefix = `HKEY_LOCAL_MACHINE\`
+
+// VerifyPath performs basic verification on registry paths.
+func VerifyPath(configurationPath string) error {
+	if !strings.HasPrefix(configurationPath, hkeyPrefix) {
+		return fmt.Errorf("invalid path [%s], only registry paths starting with %s are supported", configurationPath, hkeyPrefix)
+	}
+
+	return nil
+}
+
+func stripHKey(keypath string) (string, error) {
+	if err := VerifyPath(keypath); err != nil {
+		return "", err
+	}
+
+	return filepath.Clean(keypath[len(hkeyPrefix):]), nil
+}
+
+// ReadBinaryValue reads a REG_BINARY value from the given registry path.
+func ReadBinaryValue(keypath, valuename string) ([]byte, error) {
+	p, err := stripHKey(keypath)
+	if err != nil {
+		return nil, err
+	}
+
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, p, registry.QUERY_VALUE)
+	if err != nil {
+		return nil, fmt.Errorf("unable to open registry keypath [%s]: %v", keypath, err)
+	}
+	defer k.Close()
+
+	s, _, err := k.GetBinaryValue(valuename)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read binary (REG_BINARY) registry value [%s -> %s]: %v", p,valuename, err)
```

Comment:
Did you run gofmt? (no space between 'p' and 'valuename')?

Reply:
Done.